HelpSource/Classes/PV_Copy.schelp

```diff
@@ -17,129 +17,53 @@ source buffer.
 code::
-(
-s.waitForBoot({
-	d = Buffer.read(s, Platform.resourceDir +/+ "sounds/a11wlk01.wav");
-})
-)
-
 
-//// crossfade between original and magmul-ed whitenoise
+//proof of concept - PV_Copy creates a duplicate
 (
-x = { var in, in2, chain, chainB, chainC;
-	in = PlayBuf.ar(1, d, BufRateScale.kr(d), loop: 1) * 2;
-	in2 = WhiteNoise.ar;
-	chain = FFT(LocalBuf(2048), in);
-	chainB = FFT(LocalBuf(2048), in2);
-	chainC = PV_Copy(chain, LocalBuf(2048));
-	chainB = PV_MagMul(chainB, chainC);
-	XFade2.ar(IFFT(chain), IFFT(chainB) * 0.1, SinOsc.kr(0.1, 1.5pi));
+x = { var inA, chainA, inB, chainB, chain;
+	inA = LFClipNoise.ar(100);
+	chainA = FFT(LocalBuf(1024), inA);
+	chainB = PV_Copy(chainA, LocalBuf(1024));
+
+	IFFT(chainA) - IFFT(chainB); // cancels to zero so silent!
 }.play(s);
 )
 x.free;
 
 
-
-
-//// as previous but with Blip for 'vocoder' cross synthesis effect
 (
-x = { var in, in2, chain, chainB, chainC;
-	in = PlayBuf.ar(1, d, BufRateScale.kr(d), loop: 1) * 2;
-	in2 = Blip.ar(100, 50);
+x = { var in, chain, chainB;
+	in = WhiteNoise.ar(0.8);
 	chain = FFT(LocalBuf(2048), in);
-	chainB = FFT(LocalBuf(2048), in2);
-	chainC = PV_Copy(chain, LocalBuf(2048));
-	chainB = PV_MagMul(chainB, chainC);
-	XFade2.ar(IFFT(chain), IFFT(chainB) * 0.1, SinOsc.ar(0.1));
-}.play(s);
+	chainB = PV_Copy(chain,LocalBuf(2048));  //copies the FFT analysis into a new buffer
+	chain = PV_RandComb(chain, 0.95, Impulse.kr(0.4));
+	chainB = PV_BrickWall(chainB, SinOsc.kr(0.2));
+
+	[IFFT(chain), IFFT(chainB)*0.1];
+}.play;
 )
 x.free;
 
 
 //// Spectral 'pan'
 (
 x = { var in, chain, chainB, pan;
-	in = PlayBuf.ar(1, d, BufRateScale.kr(d), loop: 1);
+	in = BrownNoise.ar;
 	chain = FFT(LocalBuf(2048), in);
 	chainB = PV_Copy(chain, LocalBuf(2048));
 	pan = MouseX.kr(0.001, 1.001, 'exponential') - 0.001;
 	chain = PV_BrickWall(chain, pan);
 	chainB = PV_BrickWall(chainB, -1 + pan);
-	0.5 * IFFT([chain, chainB]);
-}.play(s);
-)
-x.free;
 
-
-(
-s.waitForBoot({
-	b = Buffer.alloc(s,2048,1);
-	c = Buffer.alloc(s,2048,1);
-	d = Buffer.read(s, Platform.resourceDir +/+ "sounds/a11wlk01.wav");
-	e = Buffer.alloc(s,2048,1);
-	f = Buffer.alloc(s,2048,1);
-})
-)
-
-
-//// proof of concept
-(
-x = { var inA, chainA, inB, chainB, chain;
-	inA = LFClipNoise.ar(100);
-	chainA = FFT(b, inA);
-	chainB = PV_Copy(chainA, c);
-	IFFT(chainA) - IFFT(chainB); // cancels to zero so silent!
-}.play(s);
-)
-x.free;
-// IFFTed frames contain the same windowed output data
-b.plot(\b, Rect(200, 430, 700, 300)); c.plot(\c, Rect(200, 100, 700, 300));
-
-
-
-//// Multiple Magnitude plots
-(
-x = { var in, chain, chainB, chainC;
-	in = WhiteNoise.ar;
-	chain = FFT(b, in);
-	PV_Copy(chain, LocalBuf(2048)); // initial spectrum
-	chain = PV_RectComb(chain, 20, 0, 0.2);
-	PV_Copy(chain, LocalBuf(2048)); // after comb
-	2.do({chain = PV_MagSquared(chain)});
-	PV_Copy(chain, LocalBuf(2048)); // after magsquared
-	0.00001 * Pan2.ar(IFFT(chain));
+	0.1 * IFFT([chain, chainB]);
 }.play(s);
 )
 x.free;
-
```

Comment:
is there a place where these examples still exist? They might be useful perhaps.